### ISSUE TITLE
Notification: File was not downloaded

### ISSUE THREAD

**Issue by Tobi823** [Tobi823/ffupdater#368]

**Comment by Tobi823** [comment]
On my phone only Chromium causes this background errors.

**Comment by Tobi823** [comment]
I seems that the installation works on the next background upgrade check (6-8 hours later).

**Comment by Tobi823** [comment]
Stacktrace:
```
de.marmaro.krt.ffupdater.network.exceptions.NetworkException: File was not downloaded: /storage/emulated/0/Android/data/de.marmaro.krt.ffupdater/files/Download/org.chromium.chrome.zip
	at de.marmaro.krt.ffupdater.installer.ApkChecker$Companion.throwIfDownloadedFileHasDifferentSize(ApkChecker.kt:29)
	at de.marmaro.krt.ffupdater.BackgroundJob.downloadUpdateAndReturnAvailability(BackgroundJob.kt:237)
	at de.marmaro.krt.ffupdater.BackgroundJob.access$downloadUpdateAndReturnAvailability(BackgroundJob.kt:49)
	at de.marmaro.krt.ffupdater.BackgroundJob$downloadUpdateAndReturnAvailability$1.invokeSuspend(BackgroundJob.kt:0)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:570)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:749)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:677)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:664)
```
Device information:
| Key | Value |
| --- | --- |
| FFUpdater version | 77.7.11 (137) release |
| Device | FP4 (FP4eea, FP4, FP4) |
| Manufacturer | Fairphone (Fairphone) | 
| Supported ABIs | arm64-v8a, armeabi-v7a, armeabi |
| Android version | 12 (SDK: 31) |
| OS | T2M-NB-SW-T1, android-new, release-keys, 1676445919000 |

**Comment by ledufff** [comment]
Stacktrace:
```
de.marmaro.krt.ffupdater.network.exceptions.NetworkException: File was not downloaded: /storage/emulated/0/Android/data/de.marmaro.krt.ffupdater/files/Download/us_spotco_fennec_dos_111_0_0.apk
	at de.marmaro.krt.ffupdater.installer.ApkChecker$Companion.throwIfDownloadedFileHasDifferentSize(ApkChecker.kt:29)
	at de.marmaro.krt.ffupdater.BackgroundJob.downloadUpdateAndReturnAvailability(BackgroundJob.kt:237)
	at de.marmaro.krt.ffupdater.BackgroundJob.access$downloadUpdateAndReturnAvailability(BackgroundJob.kt:49)
	at de.marmaro.krt.ffupdater.BackgroundJob$downloadUpdateAndReturnAvailability$1.invokeSuspend(BackgroundJob.kt:0)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:570)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:749)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:677)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:664)
```
Device information:
| Key | Value |
| --- | --- |
| FFUpdater version | 77.7.11 (137) release |
| Device | 21081111RG (amber_global, amber, amber) |
| Manufacturer | Xiaomi (Xiaomi) | 
| Supported ABIs | arm64-v8a, armeabi-v7a, armeabi |
| Android version | 13 (SDK: 33) |
| OS | pangu-build-component-system-30644-kkkdm-rmqw0-rn3ml, builder, release-keys, 1672846836000 |

**Comment by Tobi823** [comment]
@U-Gautam I move your bug report to the correct issue.

Stacktrace:
```
de.marmaro.krt.ffupdater.network.exceptions.NetworkException: File was not downloaded: /storage/emulated/0/Android/data/de.marmaro.krt.ffupdater/files/Download/com_brave_browser_1_49_132.apk
	at de.marmaro.krt.ffupdater.installer.ApkChecker$Companion.throwIfDownloadedFileHasDifferentSize(ApkChecker.kt:29)
	at de.marmaro.krt.ffupdater.DownloadActivity.startDownload(DownloadActivity.kt:315)
	at de.marmaro.krt.ffupdater.DownloadActivity.access$startDownload(DownloadActivity.kt:61)
	at de.marmaro.krt.ffupdater.DownloadActivity$startDownload$1.invokeSuspend(DownloadActivity.kt:0)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
	at android.os.Handler.handleCallback(Handler.java:942)
	at android.os.Handler.dispatchMessage(Handler.java:99)
	at android.os.Looper.loopOnce(Looper.java:201)
	at android.os.Looper.loop(Looper.java:288)
	at android.app.ActivityThread.main(ActivityThread.java:7884)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:936)
```
Device information:
| Key | Value |
| --- | --- |
| FFUpdater version | 77.7.14 (140) release |
| Device | Pixel 6 (oriole, oriole, oriole) |
| Manufacturer | google (Google) | 
| Supported ABIs | arm64-v8a, armeabi-v7a, armeabi |
| Android version | 13 (SDK: 33) |
| OS | abfarm-release-2004-0182, android-build, release-keys, 1677730006000 |

**Comment by Tobi823** [comment]
@DxBONESxBACK

Stacktrace:
```
de.marmaro.krt.ffupdater.network.exceptions.NetworkException: File was not downloaded: /storage/emulated/0/Android/data/de.marmaro.krt.ffupdater/files/Download/org.chromium.chrome.zip
	at de.marmaro.krt.ffupdater.installer.ApkChecker$Companion.throwIfDownloadedFileHasDifferentSize(ApkChecker.kt:29)
	at de.marmaro.krt.ffupdater.BackgroundJob.downloadUpdateAndReturnAvailability(BackgroundJob.kt:237)
	at de.marmaro.krt.ffupdater.BackgroundJob.access$downloadUpdateAndReturnAvailability(BackgroundJob.kt:49)
	at de.marmaro.krt.ffupdater.BackgroundJob$downloadUpdateAndReturnAvailability$1.invokeSuspend(BackgroundJob.kt:0)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:570)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:749)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:677)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:664)
```
Device information:
| Key | Value |
| --- | --- |
| FFUpdater version | 77.7.11 (137) release |
| Device | REVVL V+ 5G (Sprout, Sprout, S98115AA1) |
| Manufacturer | T-Mobile (Wingtech) | 
| Supported ABIs | arm64-v8a, armeabi-v7a, armeabi |
| Android version | 11 (SDK: 30) |
| OS | ubuntu, cibuild, release-keys, 1677932900000 |

**Comment by riedel** [comment]
Happens reproducibly if FFUpdater is put into the background when downloading.

**Comment by Tobi823** [comment]
> Happens reproducibly if FFUpdater is put into the background when downloading.

Thanks, that is it. I can reproduce it too. But I'm currently busy - I don't know when I can fix it.

**Comment by Tobi823** [comment]
Fixed in 78.0.1. If not, please create a new issue.
